### DISCUSSION TITLE
Veramo react native tutorial (Setup & Identifiers) on veramo.io does not work

### DISCUSSION THREAD

**Discussion by andra961** [decentralized-identity/veramo#792] [Documentation]
Hi, i'd like to point out the problems i had by following precisely each step in the [following tutorial](https://veramo.io/docs/react_native_tutorials/react_native_setup_identifers) (i use a pixel 2 android emulator).

1) The `SecretBox` import is missing (i solved with the auto import from vs code)
2) I get a warning when the app launches: `Require cycle: node_modules\react-native-crypto\index.js -> node_modules\react-native-randombytes\index.js -> node_modules\sjcl\sjcl.js -> node_modules\react-native-crypto\index.js` (didn't find a way to fix this)
3) When i press the `CREATE IDENTIFER` button in the app, i get the follwing error: `ReferenceError: Can't find variable: TextEncoder`
I fixed the error by adding: `import 'text-encoding';` in the `index.js` file of the app but i was very lucky to figure that out.

My question is: am i making some mistakes in the setup or is the the tutorial just missing the fixes i had to make?i'd also like to know if the warning can somehow be avoided or if it's just inevitable. Thank you in advance.

**Comment by mirceanis** [comment]
Thank you for reporting this!

You are right, the react native guide is a bit outdated.
The missing secret box import is a slip on our side, and the text encoding is a requirement coming from a recent update in one of our direct dependencies.

You are not doing anything wrong, quite the opposite, in fact :)

We're a very small team with a very large number of libraries under maintenance. This helps a lot!

Would you like to contribute a PR with these fixes [ to the docs](https://github.com/uport-project/veramo-website/blob/main/docs/react_native_tutorials/react_native_setup_identifiers.md) ?

**Reply by andra961** [reply]
I've submitted the PR, do you have any idea on how to fix the require cycle warning too?

**Reply by mirceanis** [reply]
Thanks for the submission!
I don't know yet how to fix the cycle warning.
I'm far from a computer for the next few days and can't check. I'll try to see about it early next year

**Reply by andra961** [reply]
thank you!
